app/components/find_interface/courses/contents_component/view.html.erb

```diff
@@ -0,0 +1,27 @@
+<h2 class="govuk-heading-m">Contents</h2>
+<ul class="govuk-list app-list--dash course-contents govuk-!-margin-bottom-8">
+  <% if about_course.present? %>
+    <li><%= govuk_link_to "Course summary", "#section-about" %></li>
+  <% end %>
+  <% if how_school_placements_work.present? || program_type == "higher_education_programme" || program_type == "scitt_programme" %>
+    <li><%= govuk_link_to course.placements_heading, "#section-schools" %></li>
+  <% end %>
+  <li><%= govuk_link_to "Entry requirements", "#section-entry" %></li>
+  <% if provider.train_with_us.present? ||  about_accrediting_body.present? %>
+    <li><%= govuk_link_to "About the training provider", "#section-about-provider" %></li>
+  <% end %>
+  <% if salaried? %>
+    <li><%= govuk_link_to "Salary", "#section-salary" %></li>
+  <% end %>
+  <li><%= govuk_link_to "Fees and financial support", "#section-financial-support" %></li>
+  <% if interview_process.present? %>
+    <li><%= govuk_link_to "Interview process", "#section-interviews" %></li>
+  <% end %>
+  <li><%= govuk_link_to "International candidates", "#section-international-students" %></li>
+  <% if provider.train_with_disability.present? %>
+    <li><%= govuk_link_to "Training with disabilities and other needs", "#section-train-with-disabilities" %></li>
+  <% end %>
+  <li><%= govuk_link_to "Contact this training provider", "#section-contact" %></li>
+  <li><%= govuk_link_to "Support and advice", "#section-advice" %></li>
+  <% if course.has_vacancies? %> <li><%= govuk_link_to "Apply", "#section-apply" %></li> <% end %>
```

Comment:
This spaghetti logic should be in view component....

...this looks like the worse code reading exercise ever